pumpkin/src/client/client_packet.rs

```diff
@@ -164,16 +164,25 @@ impl Client {
             }
         }
 
+        // Don't allow new logons when server is full.
+        // TODO: If client is an operator or otherwise suitable elevated permissions, allow client to bypass this requirement.
+        let player_count = server.get_player_count().await;
+        if player_count >= BASIC_CONFIG.max_players {
+            log::debug!("Player (IP '{}', username '{}') tried to log in, but the server is full ({}/{} players).", &self.address.lock().await, &profile.name, player_count, BASIC_CONFIG.max_players);
+            self.kick("The server is currently full, please try again later")
+                .await;
+        }
```

Comment:
Why we should check so late and even after auth ?. It would make more sense to check directly when login starts

Reply:
Thanks for the review. Would you prefer me move it to the start of `handle_login_start`?

Reply:
Yes

Reply:
Resolved in 9fdc526. 🙂